src/api/Search/types.ts

```diff
@@ -42,3 +42,76 @@ export interface SearchResponse {
 }
 
 export interface ISearchFullParams extends IGetSubGraphParams, ISearchParams {}
+
+export enum SearchSyntaxErrorKey {
+  /**
+   * 1. Error to check looping over the raw statements
+   */
+
+  // Common to all statements:
+  // When there is no value part in the statement
+  EMPTY_STATEMENT = 'empty-statement',
+
+  // Fuzzy-statement:
+  // When fuzzy value is not an integer between 1 and 100
+  INVALID_FUZZINESS = 'invalid-fuzziness',
+  // When there are more than two fuzzy statements /
+  SEVERAL_FUZZINESS = 'several_fuzziness',
+
+  // Scope-statement:
+  // When scope value is not in SearchQueryScope
+  INVALID_SCOPE = 'invalid_scope',
+  // When scope value already exists and is different
+  CONFLICTING_SCOPES = 'conflicting_scopes',
+  // When either nodes or edges
+  EDGES_NOT_SEARCHABLE = 'edges_not_searchable',
+
+  // Type-statement:
+  // When type value is comparator, range or parentheses
+  INVALID_TYPE = 'invalid-type',
+  // When type value already exists and is different
+  INCOMPATIBLE_TYPE_STATEMENTS = 'incompatible-type-statements',
+
+  // Property-statement's value
+  // Unsupported operator (more of an unsupported filter)
```

Comment:
what do you mean with (more of an unsupported filter)?

Reply:
Just a hint to remind me to throw that for filters. Hint removed